src/config/config.rs

```diff
@@ -139,17 +157,185 @@ pub struct CSIConfig {
     pub worker_port: u16,
 }
 
+impl Config {
+    /// Load config from command line, config file and environment variables
+    ///
+    /// Priority: command line > config file > environment variables
+    #[inline]
+    pub fn load() -> Result<Self, DatenLordError> {
+        // Try to load from command line
+        let arg_conf = Self::parse();
+
+        let mut builder: serfig::Builder<Self> = serfig::Builder::default();
+
+        // Load from config file
+        {
+            let config_file = match arg_conf.config_file.clone() {
+                Some(path) => path,
+                None => {
+                    if let Ok(path) = env::var("CONFIG_FILE") {
+                        path
+                    } else {
+                        String::new()
+                    }
+                }
+            };
+
+            if !config_file.is_empty() {
+                builder = builder.collect(from_file(Toml, &config_file));
+            }
+        }
+
+        // Load from environment
+        builder = builder.collect(from_env());
+
+        // Load from arguments
+        builder = builder.collect(from_self(arg_conf));
+
+        // Build the config with multi sources
+        let conf = builder.build()?;
+
+        // Check valid
+        match conf.check_valid() {
+            Ok(true) => Ok(conf),
+            Ok(false) => Err(DatenLordError::ArgumentInvalid { context: vec![] }),
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Check the config values are valid
+    #[inline]
+    pub fn check_valid(&self) -> Result<bool, DatenLordError> {
+        // Check role/node_name/node_ip/mount_path
+        let checked_values = [&self.role, &self.node_name, &self.node_ip, &self.mount_path];
+
+        let valid = checked_values.iter().all(|x| x.is_some());
+
+        Ok(valid)
+    }
```

Comment:
It seems that this function does not return any error, why do you make it return a `Result`?

Reply:
I will update it, it was initially designed to construct an exception, but later it turned out not to be very useful